### PR TITLE
[stdlib] Implement `os.path.basename`

### LINKED ISSUE #2941: [stdlib] Implement `os.path.basename`
See https://docs.python.org/3/library/os.path.html#os.path.basename for example.  This was requested internally by @abduld.

### PR DESCRIPTION
**Pull request by bgreni**
Fixes #2941

Work in progress

- Implement `os.path.basename`

### CODE REVIEW

**Comment by artemiogr97** [comment]
@bgreni FYI there was already a PR for os.path.split: https://github.com/modularml/mojo/pull/2791

**Comment by bgreni** [comment]
> @bgreni FYI there was already a PR for os.path.split: #2791

Oh sorry I missed that, I'll put this back into a draft until I can rebase on that once it lands.

**Comment by JoeLoser** [comment]
@bgreni @artemiogr97 do either one of you want to continue to get this work over the finish line?  I'd be happy to review.

**Comment by bgreni** [comment]
@JoeLoser I didn't see the other change was closed, yes I'll update this asap!

**Comment by JoeLoser** [comment]
> @JoeLoser I didn't see the other change was closed, yes I'll update this asap!

No worries, I'm just going through the old PRs and cleaning up the review queue and came across this one 😄 

Thanks!

File: stdlib/test/os/path/test_basename.mojo [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+from os.path import basename
+from pathlib import Path
+from builtin._location import __source_location
+from testing import assert_equal
+
+
+def main():
+    # Root directories
+    assert_equal("", basename("/"))
```
Comment (bgreni) [inline stdlib/test/os/path/test_basename.mojo:23]: Stole most of these from the `dirname` tests

**Review by JoeLoser — APPROVED** [review]
LGTM, thank you!

**Comment by JoeLoser** [comment]
!sync

**Comment by JoeLoser** [comment]
@bgreni do you mind adding a changelog entry?

**Pull request closed without merging**